app/src/main/java/org/qosp/notes/ui/MainActivity.kt

```diff
@@ -210,7 +215,30 @@ class MainActivity : BaseActivity() {
 
     private fun setupNavigation() {
         fun createNotebookMenuItems(notebooks: List<Notebook>) {
-            notebooks.forEach { notebook ->
+
+            // Obtaining the current setting of notebook sorting order.
+            val sort = runBlocking {
+                return@runBlocking async {
+                    preferenceRepository
+                        .getAll()
+                        .map { it.sortNavdrawerNotebooksMethod }
+                        .first()
+                        .name
+                }.await()
```

Comment:
You don't need an `async` and `await()` in the same statement. I think you can directly access the preference repository from `runBlocking`

Reply:
Done. Carried out in f949e9c4118a72110ecfe55d7de3fbf63caaedbf.